File: Go/distancecalc.go
// QeaML, 08.02.2020

package main

import (
	"fmt"
	"math"
)

func distance2d(from [2]float64, to [2]float64) float64 {
	var diffx = math.Abs(from[0] - to[0])
	var diffy = math.Abs(from[1] - to[1])
	var delta = math.Sqrt(math.Pow(diffx, 2.0) + math.Pow(diffy, 2.0))
	return delta
}

func distance3d(from [3]float64, to [3]float64) float64 {
	from2d := [2]float64{from[0], from[2]}
	to2d := [2]float64{to[0], to[2]}
	var diffx = distance2d(from2d, to2d)
	var diffy = math.Abs(from[1] - to[1])
	var delta = math.Sqrt(math.Pow(diffx, 2.0) + math.Pow(diffy, 2.0))
	return delta
}

func main() {
	a := [3]float64{1.0, 2.0, 3.0}
	b := [3]float64{4.0, 5.0, 6.0}
	fmt.Println(distance3d(a, b))
}

File: JavaScript/distanceCalc.js
function distance2d(from, to){
	diffX = Math.abs(from[0] - to[0]);
	diffY = Math.abs(from[1] - to[1]);
	delta = Math.sqrt(Math.pow(diffX, 2) + Math.pow(diffY, 2));
	return delta;
};

function distance3d(from, to){
	from2d = [from[0], from[2]];
	to2d = [to[0], to[2]];
	diffX = distance2d(from2d, to2d);
	diffY = Math.abs(from[1] - to[1]);
	delta = Math.sqrt(Math.pow(diffX, 2) + Math.pow(diffY, 2));
	return delta;
};

a = [1, 2, 3];
b = [4, 5, 6];
console.log(distance3d(a, b));